pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -215,30 +224,81 @@ public Representation post(Representation entity) {
     File tmpSegmentDir = null;
     File dataFile = null;
     try {
-
-      // 1/ Create a factory for disk-based file items
-      final DiskFileItemFactory factory = new DiskFileItemFactory();
-
-      // 2/ Create a new file upload handler based on the Restlet
-      // FileUpload extension that will parse Restlet requests and
-      // generates FileItems.
-      final RestletFileUpload upload = new RestletFileUpload(factory);
-      final List<FileItem> items;
-
-      // 3/ Request is parsed by the handler which generates a
-      // list of FileItems
-      items = upload.parseRequest(getRequest());
-
+      // 0/ Get upload type, if it's uri, then download it, otherwise, get the tar from the request.
+      Series headers = (Series) getRequestAttributes().get(RESTLET_HTTP_HEADERS);
+      String uploadTypeStr = headers.getFirstValue(FileUploadUtils.UPLOAD_TYPE);
+      FileUploadType uploadType = null;
+      try {
+        uploadType = (uploadTypeStr == null) ? FileUploadType.getDefaultUploadType() : FileUploadType.valueOf(uploadTypeStr);
+      } catch (Exception e) {
+        uploadType = FileUploadType.getDefaultUploadType();
+      }
+      String downloadURI = null;
       boolean found = false;
-      for (final Iterator<FileItem> it = items.iterator(); it.hasNext() && !found;) {
-        final FileItem fi = it.next();
-        if (fi.getFieldName() != null) {
+      switch (uploadType) {
```

Comment:
From https://github.com/linkedin/pinot/blob/master/pinot-style.xml, there is not indentation for switch/case.